lib/rspec/support/object_inspector.rb

```diff
@@ -13,6 +13,8 @@ def self.inspect(object)
           "#{object.to_s 'F'} (#{object.inspect})"
         elsif RSpec::Support.is_a_matcher?(object) && object.respond_to?(:description)
           object.description
+        elsif object.class == RSpec::Mocks::Double
```

Comment:
We also can't count on `RSpec::Mocks::Double` being defined.  (For example, when rspec-expectations is used on a project but not rspec-mocks).  So this would have to be `defined?(RSpec::Mocks::Double) && RSpec::Mocks::Double === object`